python_modules/dagster/dagster/_core/definitions/source_asset.py

```diff
@@ -104,6 +116,7 @@ def __init__(
         group_name: Optional[str] = None,
         resource_defs: Optional[Mapping[str, ResourceDefinition]] = None,
         observe_fn: Optional[SourceAssetObserveFunction] = None,
+        _required_resource_keys: Optional[AbstractSet[str]] = None,
```

Comment:
I think worth mentioning why this has an underscore - i.e. that it's used for the observation function, but we're not sure what we want to do when an asset is both observable and can also be subject to other operations.